packages/fluentui/react/src/components/Toolbar/ToolbarMenu.tsx

```diff
@@ -45,75 +58,73 @@ export interface ToolbarMenuProps extends UIComponentProps, ChildrenComponentPro
   submenuIndicator?: ShorthandValue<BoxProps>;
 }
 
-class ToolbarMenu extends UIComponent<ToolbarMenuProps> {
-  static displayName = 'ToolbarMenu';
-
-  static className = 'ui-toolbar__menu';
+export type ToolbarMenuStylesProps = never;
 
-  static create: ShorthandFactory<ToolbarMenuProps>;
+const ToolbarMenu: React.FC<WithAsProp<ToolbarMenuProps>> & FluentComponentStaticProps = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(ToolbarMenu.displayName, context.telemetry);
+  setStart();
 
-  static propTypes = {
-    ...commonPropTypes.createCommon(),
-    items: customPropTypes.collectionShorthandWithKindProp(['divider', 'item']),
-    onItemClick: PropTypes.func,
-    submenu: PropTypes.bool,
-    submenuIndicator: customPropTypes.shorthandAllowingChildren
-  };
+  const { accessibility, className, children, design, items, submenu, submenuIndicator, styles, variables } = props;
 
-  static defaultProps = {
-    accessibility: toolbarMenuBehavior,
-    as: 'ul'
-  };
-
-  actionHandlers = {
-    performClick: e => {
-      _.invoke(this.props, 'onClick', e, this.props);
-    }
-  };
+  const getA11yProps = useAccessibility(accessibility, {
+    debugName: ToolbarMenu.displayName,
+    actionHandlers: {
+      performClick: e => {
+        _.invoke(props, 'onClick', e, props);
+      }
+    },
+    rtl: context.rtl
+  });
+  const { classes } = useStyles<ToolbarMenuStylesProps>(ToolbarMenu.displayName, {
+    className: ToolbarMenu.className,
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables
+    }),
+    rtl: context.rtl
+  });
 
-  handleItemOverrides = variables => predefinedProps => ({
+  const handleItemOverrides = predefinedProps => ({
     onClick: (e, itemProps) => {
       _.invoke(predefinedProps, 'onClick', e, itemProps);
-      _.invoke(this.props, 'onItemClick', e, {
+      _.invoke(props, 'onItemClick', e, {
         ...itemProps,
         menuOpen: !!itemProps.menu
       });
     },
     variables: mergeComponentVariables(variables, predefinedProps.variables)
   });
 
-  handleDividerOverrides = variables => predefinedProps => ({
+  const handleDividerOverrides = predefinedProps => ({
     variables: mergeComponentVariables(variables, predefinedProps.variables)
```

Comment:
A devil is here 😈 Previously we merged resolved variables while now we merge input, the simplified change:

```diff
-renderComponent({ variables }) {
+  const { variables } = this.props
}
```

But as I discussed this with @miroslavstastny, it's valid. As we want to pass down only overrides.